src/libUtils/EvmUtils.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2022 Zilliqa
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <array>
+#include <iostream>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+
+#include <boost/filesystem.hpp>
+#include "EvmUtils.h"
+
+#include "JsonUtils.h"
+#include "Logger.h"
+#include "common/Constants.h"
+#include "libData/AccountData/Account.h"
+#include "libData/AccountData/TransactionReceipt.h"
+#include "libPersistence/ContractStorage.h"
+#include "libUtils/EvmCallParameters.h"
+#include "libUtils/EvmJsonResponse.h"
+
+using namespace std;
+using namespace boost::multiprecision;
+
+Json::Value EvmUtils::GetEvmCallJson(const EvmCallParameters& params) {
+  Json::Value arr_ret(Json::arrayValue);
+
+  arr_ret.append(params.m_contract);
+  arr_ret.append(params.m_caller);
+  std::string code;
+  try {
+    // take off the EVM prefix
+    std::copy(params.m_code.begin() + 3, params.m_code.end(),
+              std::back_inserter(code));
+    arr_ret.append(code);
+  } catch (std::exception& e) {
+    arr_ret.append(params.m_code);
```

Comment:
error handling seems wrong, we can't assume all is good  if we can't chop off 'EVM'

Reply:
Better error handling and reporting of what decision was taken at runtime,

This should be discussed internally on the project, for now it reports and sends onwards assuming it was intentional by caller.